src/assets/build_assets/acknowledgements.rs

```diff
@@ -0,0 +1,199 @@
+use std::fs::read_to_string;
+use std::path::Path;
+
+use crate::error::*;
+
+// Sourced from the License section in the README.md
+const PREAMBLE: &str = "Copyright (c) 2018-2021 bat-developers (https://github.com/sharkdp/bat).
+
+bat is made available under the terms of either the MIT License or the Apache
+License 2.0, at your option.
+
+See the LICENSE-APACHE and LICENSE-MIT files for license details.
+";
+
+/// Looks for LICENSE and NOTICE files in `source_dir`, does some rudimentary
+/// analysis, and compiles them together in a single string that is meant to be
+/// used in the output to `--acknowledgements`
+pub fn build_acknowledgements(source_dir: &Path, acknowledgements: bool) -> Result<Option<String>> {
+    if !acknowledgements {
+        return Ok(None);
+    }
+
+    let mut acknowledgements = String::new();
+    acknowledgements.push_str(PREAMBLE);
+
+    // Sort entries so the order is stable over time
+    let entries = walkdir::WalkDir::new(source_dir).sort_by(|a, b| a.path().cmp(b.path()));
+    for entry in entries {
+        let entry = match entry {
+            Ok(entry) => entry,
+            Err(_) => continue,
+        };
+
+        let path = entry.path();
+        let stem = match path.file_stem().and_then(|s| s.to_str()) {
+            Some(stem) => stem,
+            None => continue,
+        };
+
+        handle_file(&mut acknowledgements, path, stem)?
+    }
+
+    Ok(Some(acknowledgements))
+}
+
+fn handle_file(acknowledgements: &mut String, path: &Path, stem: &str) -> Result<()> {
+    if stem == "NOTICE" {
+        handle_notice(acknowledgements, path)?;
+    } else if stem.to_ascii_uppercase() == "LICENSE" {
+        handle_license(acknowledgements, path)?;
+    }
+
+    Ok(())
+}
+
+fn handle_notice(acknowledgements: &mut String, path: &Path) -> Result<()> {
+    // Assume NOTICE as defined by Apache License 2.0. These must be part of acknowledgements.
+    let license_text = read_to_string(path)?;
+    append_to_acknowledgements(acknowledgements, &license_text)
+}
+
+fn handle_license(acknowledgements: &mut String, path: &Path) -> Result<()> {
+    let license_text = read_to_string(path)?;
+
+    if include_license_in_acknowledgments(&license_text) {
+        append_to_acknowledgements(acknowledgements, &license_text)
```

Comment:
I wonder if it could make sense to append the path to the output too, to make it clear where each license/notice comes from, rather than having no context.

Reply:
Good input. I will adjust the code for that. Seems like a good thing to do.

Reply:
Fixed now (I will soon make a comment that shows how it look)